### ISSUE TITLE
`let default` results in keyword `let` being reported as reserved

### ISSUE THREAD

**Issue by Alaneor** [eslint/espree#313]
The following code:

```js
espree.parse('let default = true', { ecmaVersion: 6, sourceType: 'module' })
```

Results in the following error: `SyntaxError: The keyword 'let' is reserved`

**Expected behaviour**: I would have expected to see the `default` being reported as reserved keyword, not `let`.

When `sourceType: 'script'` or when I use `const` or `var` instead of `let`, the error seems to match my expectations by reporting the following: `SyntaxError: Unexpected token default`

**Comment by ilyavolodin** [comment]
What version of Espree are you using? This has been fixed a long time ago. Here's the related issue: https://github.com/eslint/espree/issues/149

**Comment by not-an-aardvark** [comment]
@ilyavolodin I think https://github.com/eslint/espree/issues/149 might be a different issue. `let default = 0` is invalid syntax -- the issue here is that the error message is wrong (it complains about `let` instead of `default`).

**Comment by Alaneor** [comment]
I have 3.3.2 installed. Looks like that's the latest.

**Comment by ilyavolodin** [comment]
Ah, sorry, @not-an-aardvark is correct. I misread the issue. I think this would be an issue upstream in Acorn.

**Comment by robertrossmann** [comment]
Opened a new issue in Acorn repo: https://github.com/ternjs/acorn/issues/544 🎉

Feel free to close this unless you would like to keep track of this. Thanks!
